docs-src/api-overview.md

```diff
@@ -79,6 +96,132 @@ BackendPolicy. For more information on what may be configured with this resource
+- **Kind** - A Gateway listener can only select a single type of Route 
+resource. This could be an HTTPRoute, TCPRoute, or a custom Route type.
+- **Namespace** - A Gateway can also control from which Namespaces Routes can be
+selected via the `namespaces.from` field. It supports three possible values:
+    - `SameNamespace` is the default option. Only Routes in the same namespace 
+    as this Gateway will be selected. 
+    - `All` will select Routes from all Namespaces.
+    - `Selector` means that Routes from a subset of Namespaces selected by a 
+    Namespace label selector will be selected by this Gateway. When `Selector` 
+    is used then the `listeners.routes.namespaces.selector` field can be used to 
+    specify label selectors. This field is not supported with `All` or 
+    `SameNamespace`.
+
+The below Gateway will select all HTTPRoute resources with the `gateway:
+prod-web-gw` across all Namespaces in the cluster.
+
+```
+kind: Gateway
+...
+spec:
+  listeners:  
+  - routes:
+      kind: HTTPRoute
+      selector:
+        matchLabels:
+          gateway: prod-web-gw 
+      namespaces:
+        from: All
+```
+
+#### Gateway Selection
+
+Routes can also select which Gateways they can select. The `gateways.allow`
+field supports three values:
+
+- `All` is the default value if none is specified. This leaves all binding 
+to the Route label and Namespace selectors on the Gateway. 
+- `SameNamespace` only allows this Route to bind with Gateways from the 
+same Namespace.
+- `FromList` allows an explicit list of Gateways to be specifiied that a 
+Route will bind with. `gateways.gatewayRefs` is only supported with this option. 
+
+The following `my-route` Route selects only the `foo-gateway` in the
+`foo-namespace` and will not be able to bind with any other Gateways. Note that
+`foo-gateway` is in a different Namespace. If the `foo-gateway` allows
+cross-Namespace binding and also selects this Route then `my-route` will bind to
+it. 
+
+```yaml
+kind: HTTPRoute
+metadata:
+  name: my-route
+  namespace: bar-namespace
+spec:
+  gateways:
+    allow: FromList
+    gatewayRefs:
+    - name: foo-gateway
+      namespace: foo-namespace
+```
+
+Note that Gateway and Route binding is bi-directional. This means that both
+resources must select each other for them to bind. If a Gateway has Route label
+selectors that do not match any existing Route then nothing will bind to it even
+if a Route's `spec.gateways.allow = All`. Similarly, if a Route references a
+specific Gateway, but that Gateway is not selecting the Route's Namespace, then
+they will not bind. A binding will only take place if both resources select each
+other.
+
+It may not always be apparent from the resource specifications which Gateways
+and Routes are bound, but binding can be determined from the resource status.
+The [Route status](httproute.md#routestatus) will list all of the Gateways that
+a Route is bound to and any relevant conditions for the binding.
```

Comment:
This is a tiny nitpick that may not be worth adding, but technically it's the first 100 Gateways that select a Route. (https://github.com/kubernetes-sigs/service-apis/blob/master/apis/v1alpha1/shared_types.go#L198-L202)

Reply:
I'll leave this detail to the HTTPRoute page since this is still the concepts page.